lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -122,6 +125,46 @@ def render_pson(catalog)
       pson
     end
 
+    def clean_sensitive_parameters(catalog)
+      catalog['resources'].map! do |resource|
+        if resource.key? 'sensitive_parameters'
+          resource['sensitive_parameters'].each do |p|
+            hash = Digest::SHA256.hexdigest Marshal.dump(resource['parameters'][p])
+            resource['parameters'][p] = 'Sensitive [hash ' + hash + ']'
+          end
+          resource.tap { |hs| hs.delete('sensitive_parameters') }
+        else
+          resource
+        end
+      end
+
+      catalog
+    end
+
+    def clean_nested_sensitive_parameters(catalog)
+      # Resources can also contain sensitive data nested deep in hashes/arrays
+      catalog['resources'].each do |resource|
+        redact_sensitive(resource['parameters']) if resource.key? 'parameters'
+      end
+
+      catalog
+    end
+
+    def redact_sensitive(data)
```

Comment:
Could this be used in `clean_sensitive_parameters` to reduce code duplication?

Reply:
Not sure...